File: python/completed/euler0063.py
###
#
# Created by Holden on 12/8/2017
#
# COMPLETED on 12/10/2017
#
# Problem 63 - Powerful Digits Count
#
###

import math


def powerful_digits_count():
    powerful = []
    base = 1
    power = 1
    temp = base ** power

    while base < 10:
        while (int(math.log10(temp)) + 1) >= power:
            if (int(math.log10(temp)) + 1) == power:
                powerful.append([base, power, temp])
            power += 1
            temp = base ** power

        base += 1
        power = 1
        temp = base ** power
    print(powerful)

    return len(powerful)


print(powerful_digits_count())

File: javascript/completed/euler0063.js
export function solve63() {
  let count = 0;
  for (let base = 0; base <= 9; base++) {
    for (let pow = 0; pow <= 20; pow++) {
      const result = Math.pow(base, pow);
      if (result.toString().length === pow) {
        count++;
      }
    }
  }

  return count;
}
